### ISSUE TITLE
Is ':' in concept:name not allowed? 

### ISSUE THREAD

**Issue by treand** [process-intelligence-solutions/pm4py#344]
Hello,

 

while working with heuristics nets I discovered that if the strings of 'concept:name' contain ':' the resulting figure will be not complete.
Below a small example. I am using pm4py version '2.2.25'.

```
import numpy as np
from pm4py.algo.discovery.heuristics import algorithm as heuristics_miner
from pm4py.visualization.heuristics_net import visualizer as hn_visualizer

dummy = pd.DataFrame(np.array([
['A', '06/10/2021' , 'Begin gettingup' ],
['A', '06/11/2021' , 'eating'],
['A', '06/12/2021' , 'sleeping'],
['B', '06/12/2021' , 'Begin gettingup'],
['B', '06/13/2021' , 'eating'],
['B', '06/13/2021' , 'singing'],
['B', '06/14/2021' , 'sleeping'],
['C', '06/06/2021' , 'Begin gettingup'],
['C', '06/11/2021' , 'sleeping'],
['C', '06/12/2021' , 'eating'],
['D', '06/15/2021' , 'Begin yawning'],
['D', '06/16/2021' , 'eating'],
['E', '06/15/2021' , 'Begin yawning'],
['E', '06/17/2021' , 'eating'],
['E', '06/26/2021' , 'eating'],
['F', '06/11/2021' , 'Begin gettingup'],
['F', '06/15/2021' , 'sleeping']]),
columns=['case:concept:name', 'time:timestamp', 'concept:name'])

log = log_converter.apply(dummy)
heu_net = heuristics_miner.apply_heu(log)
gviz = hn_visualizer.apply(heu_net)
hn_visualizer.view(gviz)
```
The resulting figure:
<img width="248" alt="f1" src="https://user-images.githubusercontent.com/37930704/183040571-e223ed71-bb7b-45b0-9d49-1c116c0ad8f1.png">

 
Then adding ':' to the starting activity:
```
dummy = pd.DataFrame(np.array([
['A', '06/10/2021' , 'Begin: gettingup' ],
['A', '06/11/2021' , 'eating'],
['A', '06/12/2021' , 'sleeping'],
['B', '06/12/2021' , 'Begin: gettingup'],
['B', '06/13/2021' , 'eating'],
['B', '06/13/2021' , 'singing'],
['B', '06/14/2021' , 'sleeping'],
['C', '06/06/2021' , 'Begin: gettingup'],
['C', '06/11/2021' , 'sleeping'],
['C', '06/12/2021' , 'eating'],
['D', '06/15/2021' , 'Begin: yawning'],
['D', '06/16/2021' , 'eating'],
['E', '06/15/2021' , 'Begin: yawning'],
['E', '06/17/2021' , 'eating'],
['E', '06/26/2021' , 'eating'],
['F', '06/11/2021' , 'Begin: gettingup'],
['F', '06/15/2021' , 'sleeping']]),
columns=['case:concept:name', 'time:timestamp', 'concept:name'])

log = log_converter.apply(dummy)
heu_net = heuristics_miner.apply_heu(log)
gviz = hn_visualizer.apply(heu_net)
hn_visualizer.view(gviz)
```
And the resulting figure:
<img width="131" alt="f2" src="https://user-images.githubusercontent.com/37930704/183040630-594b6afb-e61a-4845-b4f4-b0ea4b64e672.png">

Thanks in advance for your help :)

**Comment by Javert899** [comment]
Dear treand,

Thanks for signaling the problem. I can anticipate the problem is in the visualizer (pydotplus) that we use for heuristics net. It does not like the : very much in the node names. We identified a solution for it and come back to you when it will be released.

**Comment by fit-alessandro-berti** [comment]
Dear treand, the issue has been resolved in pm4py 2.2.26

Thanks for the collaboration and good luck for your projects!

**Comment by treand** [comment]
Thank you for the quick fix!
